lib/logic/bloc/stats_bloc/stats_bloc.dart

```diff
@@ -363,7 +364,7 @@ class StatsBloc extends Bloc<StatsEvent, StatsState> {
     for (var year in years) {
       final booksInyear = List<Book>.empty(growable: true);
       for (Book book in books) {
-        if (book.finishDate != null && book.rating != null) {
+        if ((book.finishDate != null || book.readingTime != null) && book.rating != null) {
           final finishYear = book.finishDate!.year;
```

Comment:
I don't think this will work - book.finishDate is not guaranteed to be non null now

Reply:
I don't think this is resolved - the statement is true if finishDate != null or readingTime != null. So there's no guarantee that finishDate is non null everytime.

Line
final finishYear = book.finishDate!.year;
Will throw an exception if finishDate is null.